src/main/java/step3/domain/CarRacingGame.java

```diff
@@ -29,11 +32,10 @@ private void playAllRounds(RacingCars cars) {
     private void playOneRound(RacingCars cars) {
         Round round = new Round();
         round.start(cars, gameSetting);
-
-        playedRounds.add(round);
+        this.gameResult.getPlayedRounds().add(round);
```

Comment:
이곳에서, 9번째 줄에 정의한 CarRacingGameResult안에 감싸져 있는 List<Round>를 꺼낸후,  해당   List<Round>에 새로운 요소를 더하고 있습니다.

즉, 9번쨰 줄에 있는 객체의 내용을 계속 변경하면서 재사용하고 있는 셈입니다.
제가 찾아보니 9번쨰 줄에 있는 객체를 재사용하지 않고, 값이 추가될때 마다 새로운 객체를 9번째줄 객체에 할당하는 방식도 있는것 같았습니다. 즉 35번쨰 줄을 이렇게 바꿀수도 있을것 같습니다.

`
 List<Round> playedRounds = gameResult.getPlayedRounds();
playedRounds.add(round);
 gameResult = new CarRacingGameResult(playedRounds);
`

**기존 객체 재사용 vs 매번 새로운 객체를 생성해 재할당**

준일님은 어떤방식이 좋다고 생각하시나요?

Reply:
일단 getter로 받아온 컬렉션에 다시 add 같은 메소드를 통해서 값을 변형시킬 경우 문제가 될 수 있습니다. 캡슐화의 원칙이 깨지기 때문에 객체지향을 사용하는 목적이 불분명해지죠
```suggestion
        this.gameResult.addRound(round);
```
그래서 이렇게 메소드를 하나 더만들어서 관리하는게 좋답니다.

> 기존 객체 재사용 vs 매번 새로운 객체를 생성해 재할당

매번 새로운 객체를 만드는 방법이 더 좋답니다! 그래야 예측 가능한 코드를 만들 수 있어요~

Reply:
그렇군요! Getter로 받아서 다시 add를 이용하는 방식은 지양하도록 하겠습니다.
다음 미션에서는 알려주신대로 라운드를 추가하는 행동(메서드)을 일급 컬렉션에  넣어서 처리하도록 해보겠습니다.
깔끔하게 바뀔 코드가 기대되네요!